imap_processing/ialirt/process_ephemeris.py

```diff
@@ -0,0 +1,199 @@
+        Doppler shift. Currently a throwaway value.
+    """
+    return 1
+
+
+@typing.no_type_check
+@ensure_spice
+def latitude_longitude_to_ecef(
+    longitude: float, latitude: float, altitude: float
+) -> ndarray:
+    """
+    Convert geodetic coordinates to rectangular coordinates.
+
+    Earth-Centered, Earth-Fixed (ECEF) coordinates are a Cartesian coordinate system
+    with an origin at the center of the Earth.
+
+    Parameters
+    ----------
+    longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian, negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative to south.
+    altitude : float
+        Altitude in kilometers.
+
+    Returns
+    -------
+    rect_coords : ndarray
+        Rectangular coordinates in kilometers.
+    """
+    latitude_radians = np.deg2rad(latitude)
+    longitude_radians = np.deg2rad(longitude)
+
+    # Retrieve Earth's radii from SPICE
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.
+    # (url cont.) bodvrd
+    radii = spice.bodvrd("EARTH", "RADII", 3)[1]
+    equatorial_radius = radii[0]  # Equatorial radius in km
+    polar_radius = radii[2]  # Polar radius in km
+    flattening = (equatorial_radius - polar_radius) / equatorial_radius
+
+    # Convert geodetic coordinates to rectangular coordinates
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.
+    # (url cont.) georec
+    rect_coords = spice.georec(
+        longitude_radians, latitude_radians, altitude, equatorial_radius, flattening
+    )
+
+    return rect_coords
+
+
+@typing.no_type_check
+@ensure_spice
+def calculate_azimuth_and_elevation(
+    longitude: float,
+    latitude: float,
+    altitude: float,
+    observation_time: float,
+    target: str = "IMAP",
+) -> tuple:
+    """
+    Calculate azimuth and elevation.
+
+    Parameters
+    ----------
+    longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
+    altitude : float
+        Altitude in kilometers.
+    observation_time : float
+        Time at which the state of the target relative to the observer
+        is to be computed. Expressed as ephemeris time, seconds past J2000 TDB.
+    target : str (Optional)
+        The target body. Default is "IMAP".
+
+    Returns
+    -------
+    azimuth : float
+        Azimuth in degrees.
+    elevation : float
+        Elevation in degrees.
+    """
+    observer_position_ecef = latitude_longitude_to_ecef(longitude, latitude, altitude)
+
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.
+    # (url cont.) azlcpo
+    azel_results = spice.azlcpo(
+        method="Ellipsoid",  # Only method supported
+        target=target,  # target ephemeris object
```

Comment:
If you use a `SpiceBody` for the target input type, this will need to be:
```suggestion
        target=target.name,  # target ephemeris object
```